### DISCUSSION TITLE
Using driver API directly

### DISCUSSION THREAD

**Discussion by henryruhs** [liquidctl/liquidctl#573] [General]
I like to use specific drivers for an integration:

```python
#!/usr/bin/env python

from liquidctl.driver.rgb_fusion2 import RgbFusion2

devs = RgbFusion2.find_supported_devices()

for dev in devs:
	with dev.connect():
		print(f'{dev.description} at {dev.bus}:{dev.address}:')
		dev.set_color('sync', 'fixed', [(255, 255, 0)])
```

Is this considered to be safe in terms of a public and stable API or more like internals that can break anytime?

**Comment by jonasmalacofilho** [comment]
Yes, `RgbFusion2.find_supported_devices()` is stable, according to our [stability pledge](https://github.com/liquidctl/liquidctl/blob/main/docs/developer/process.md#stability-and-backward-compatibility).

And, more generally, as long as `<DriverClass>` isn't documented otherwise, `<DriverClass>.find_supported_devices()` is stable:

https://github.com/liquidctl/liquidctl/blob/dd2d1c1cf18d3ecd7c4bff0c01ed52112e45b1c4/liquidctl/driver/__init__.py#L11-L15

Admittedly that mention in the documentation is hard to find.
